File: 006_ruby_Counting_sheep/count_sheep.rb
# origin: https://www.codewars.com/kata/54edbc7200b811e956000556
def countSheeps(array)
    sheep_counter = 0
    array.each do |sheep| 
      sheep == true ? sheep_counter += 1 : 0
    end
    return sheep_counter
end

# alternative solution:
def countSheeps array
    array.count(true)
end

File: 87_typescript_countSheep/countSheep.ts
// origin: https://www.codewars.com/kata/if-you-cant-sleep-just-count-sheep/train/typescript
export function countSheep(num: number): string {
	const result: string[] = [];
	for (let i = 1; i <= num; i++) {
		result.push(`${i} sheep...`);
	}
	return result.join('');
}

// alternative solution
export function countSheepAlt(num: number): string {
	return Array.from({ length: num }, (x, i) => `${i + 1} sheep...`).join('');
}
